STAR-151 End to End Testing - Playwright

susanssky opened the pull request:
- Issue #4

- Refer to the `README.md` in the PR files

susanssky committed:
fixed the id of TextareaAutosize

susanssky committed:
add "data-testid" to the 2 icons

susanssky committed:
inital config

bazmurphy commented:
Can we move all of it into a single folder, we are trying to avoid having any package.json node_modules etc at the root

Maybe call it /integration/ and move all the stuff there... and do we need tests/e2e subfolder or can it be simplified?

bazmurphy commented:
0. typescript should be installed at a per folder level
your typescript installation on your local machine is GLOBAL, this is not best practice
I would advise you to remove your global typescript installation 
and instead get in the habit of doing localised typescript installs
0b. there is no typescript configuration file

1. move all the logic into a single folder, we are trying to keep the root clean

2. better to use an npm script `npm start` that runs the `npx playwright test` command

3. playwright has to install various things before it can run:
![](https://github.com/star-cyf/star/assets/61154071/7d129083-004e-411d-b93d-0f9d0c53318f)
![](https://github.com/star-cyf/star/assets/61154071/3315ae14-da27-4541-ab24-0c3839bde7fb)

4. google has to use 2FA before continuing each time the test wil run (how can we prevent this? make a google account to use as a tester and all share the credentials?)
![image](https://github.com/star-cyf/star/assets/61154071/0aaf839f-41f8-45d8-a8d9-08f23585c892)

5. `client` and `server` both have to be up and running in dev mode - could this be built into the script (not sure?)

6. typing issues process and other because typescript is not reading from the devDependencies @types you have

I will have a look and play now and discuss the fixes after...

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

susanssky committed:
fix the issue of not being able to find the newest question if questions > 5or10? in the testing

susanssky committed:
fix the manual issue of storage-state.json

fix the issue where the user needs to re-login during tests if storage-state.json exists and is not expired, and manually uncomment the globalSetup property in playwright.config.ts if it does not exist or is expired

susanssky committed:
push the firefox login way to consider

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

susanssky committed:
create teardown

susanssky committed:
Merge branch 'STAR-151' of https://github.com/star-cyf/star into STAR-151

bazmurphy committed:
fix package-lock.json

bazmurphy committed:
fix prettier command

bazmurphy committed:
fix server workflow lint command

susanssky committed:
Refactor the question page tests using Page object models and create tests for the questions page

susanssky committed:
Merge branch 'STAR-151' of https://github.com/star-cyf/star into STAR-151

susanssky committed:
fix issue of chrome login

susanssky committed:
create testing of Home

susanssky committed:
fix typo

bazmurphy committed:
update readme

susanssky committed:
add parallel model

susanssky committed:
working...

susanssky committed:
working

susanssky committed:
complete testing of profilePage & sort feature

susanssky committed:
add search testing

susanssky committed:
a bit updating

susanssky committed:
a bit updating

bazmurphy committed:
fix formatDate

susanssky committed:
updated the files of non e2e-testing

bazmurphy committed:
adjustments & cleanup

bazmurphy committed:
cleanup adjustment

susanssky committed:
update playwright .config.ts

susanssky committed:
update playwright .config.ts

bazmurphy committed:
suppress logging in testing environment

bazmurphy committed:
supress seed intermediate seed logs

bazmurphy committed:
supress seed intermediate seed logs 2

susanssky committed:
a bit update

susanssky committed:
add workflow for tests-e2e

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy committed:
adjustments

bazmurphy reviewed (approved):
Fantastic Job 😎
End to End testing within CI/CD
Very professional 👍
